pennylane/workflow/qnode.py

```diff
@@ -777,25 +734,19 @@ def get_best_method(device, interface, tape=None):
             tuple[str or .TransformDispatcher, dict, .Device: Tuple containing the ``gradient_fn``,
             ``gradient_kwargs``, and the device to use when calling the execute function.
         """
+        if isinstance(device, qml.devices.LegacyDevice):
+            device = qml.devices.LegacyDeviceFacade(device)

```

Comment:
Another problematic instance check. I'm starting to wonder if it even makes sense to include the metaclass at this point.

Reply:
Discussed offline. Decided this is fine given the idempotency of the Facade constructor.